ament_lint_cmake/ament_lint_cmake/cmakelint.py

```diff
@@ -34,7 +34,7 @@
 _RE_COMMAND_END_SPACES = re.compile(r'(\s*)\)', re.VERBOSE)
 _RE_LOGIC_CHECK = re.compile(r'(\w+)\s*\(\s*\S+[^)]+\)', re.VERBOSE)
 _RE_COMMAND_ARG = re.compile(r'(\w+)', re.VERBOSE)
-_RE_STRING = re.compile(r'^\s*#?\s*"[^"]+"\)?', re.VERBOSE)
+_RE_STRING = re.compile(r'^\s*\#?\s*"[^"]+"\)?', re.VERBOSE)
```

Comment:
so this is matching:

```
<whitespace_character>* <pound_sign>? <whitespace_character>* <non_empty_quoted_string> <interrogation_mark>?
```

(hope my pseudo-regex syntax is clear)

I don't get why accepting an optional pound sign there, though the previous regex expression was definitely broken.

Reply:
> I don't get why accepting an optional pound sign there
I think it's there so that the expression still matches a line that's commented out.
Some examples that will be caught because of this:
```cmake
#message(STATUS "Pretend this is a really long line)
if(FOO)
  # message(STATUS "Pretend this is ALSO a really long line)
endif()
```

Reply:
Also, I think the bit at the end of the regex matches an optional `)` character.

Reply:
> I think it's there so that the expression still matches a line that's commented out.

Ah, makes sense!

> Also, I think the bit at the end of the regex matches an optional ) character.

Yeap, I realized but wrote something else :joy: